homeassistant/components/camera/synology.py

```diff
@@ -203,48 +92,48 @@ def camera_image(self):
     @asyncio.coroutine
     def async_camera_image(self):
         """Return a still image response from the camera."""
-        image_url = SYNO_API_URL.format(
-            self._synology_url, WEBAPI_PATH, self._camera_path)
-
-        image_payload = {
-            'api': CAMERA_API,
-            'method': 'GetSnapshot',
-            'version': '1',
-            'cameraId': self._camera_id
-        }
-        try:
-            with async_timeout.timeout(self._timeout, loop=self.hass.loop):
-                response = yield from self._websession.get(
-                    image_url,
-                    params=image_payload
-                )
-        except (asyncio.TimeoutError, aiohttp.ClientError):
-            _LOGGER.error("Error fetching %s", image_url)
-            return None
-
-        image = yield from response.read()
-
-        return image
+        return self._surveillance.get_camera_image(self._camera_id)
 
     @asyncio.coroutine
     def handle_async_mjpeg_stream(self, request):
         """Return a MJPEG stream image response directly from the camera."""
-        streaming_url = SYNO_API_URL.format(
-            self._synology_url, WEBAPI_PATH, self._streaming_path)
-
-        streaming_payload = {
-            'api': STREAMING_API,
-            'method': 'Stream',
-            'version': '1',
-            'cameraId': self._camera_id,
-            'format': 'mjpeg'
-        }
-        stream_coro = self._websession.get(
-            streaming_url, params=streaming_payload)
+        streaming_url = self._camera.video_stream_url
+        stream_coro = self._websession.get(streaming_url)
 
         yield from async_aiohttp_proxy_web(self.hass, request, stream_coro)
 
     @property
     def name(self):
         """Return the name of this device."""
-        return self._name
+        return self._camera.name
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return self._camera.is_recording
+
+    def should_poll(self):
+        """Update the recording state periodically."""
+        return True
+
+    @asyncio.coroutine
+    def async_update(self):
```

Comment:
It is not async

Reply:
Need your clarification on this

I see there's update() and async_update(), but I couldn't find where update() is called.
The only thing I found is async_update() is called in camera/__init__.py when polling is enabled:

```
    @asyncio.coroutine
    def async_handle_camera_service(service):
        """Handle calls to the camera services."""
        target_cameras = component.async_extract_from_service(service)

        for camera in target_cameras:
            if service.service == SERVICE_EN_MOTION:
                yield from camera.async_enable_motion_detection()
            elif service.service == SERVICE_DISEN_MOTION:
                yield from camera.async_disable_motion_detection()

        update_tasks = []
        for camera in target_cameras:
            if not camera.should_poll:
                continue

            update_coro = hass.async_add_job(
                camera.async_update_ha_state(True))
            if hasattr(camera, 'async_update'):
                update_tasks.append(update_coro)
            else:
                yield from update_coro

        if update_tasks:
            yield from asyncio.wait(update_tasks, loop=hass.loop)
```

I see there's a comment in camera.nest that the update doesn't seem to be called:
https://github.com/home-assistant/home-assistant/blob/2df433eb0a11752a8cb7b5bedbe87296539c801f/homeassistant/components/camera/nest.py#L74
I debugged it a bit and can confirm that I never seen update() being called when I left nest cam running for more than a day.

Reply:
I tried again with update() and looks like I was wrong, it did get called. So I've moved code in async_update() to update() now.

Reply:
But I still couldn't figure out where update() gets called in the code, would you point me where it is?

Reply:
helpers/entity.py

Reply:
Got it, thanks

Just for my reference: https://github.com/home-assistant/home-assistant/blob/2df433eb0a11752a8cb7b5bedbe87296539c801f/homeassistant/helpers/entity.py#L215